### DISCUSSION TITLE
Using rg parts as crate, redirecting output

### DISCUSSION THREAD

**Discussion by bq-wrongway** [BurntSushi/ripgrep#2433] [Q&A] [answered]
Hi,

i would like to add grep / grep_searcher to my small app, but instead of providing sink to printout the result i would like result to be directly redirected and saved to a given file : 
```

let result = searcher.search_path(
                &matcher,
                path_provider.path(),
                printer.sink(&matcher),
            );
```
Is it possible to achieve something like this ?
If there is a match auto save it to file
if not nothing

**Comment by BurntSushi** [comment, marked as answer]
Well have you tried calling [`Standard::new`](https://docs.rs/grep-printer/latest/grep_printer/struct.Standard.html#method.new) with a `std::io::BufReader<std::fs::File>`?

**Reply by bq-wrongway** [reply]
Thanks,i managed to get it working with your help!
